docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+
+### Router Cache
+
+Next.js has an in-memory client-side cache that stores the React Server Component Payload, split by individual routes. This **Router Cache** is used to improve the navigation experience. Next.js [prefetches](/docs/app/building-your-application/routing/linking-and-navigating#prefetching) and caches routes that the user is likely going to navigate to (when using the `<Link >` component), and caches visited route segments, enabling [partial rendering](/docs/app/building-your-application/routing/linking-and-navigating#partial-rendering). This applies to newly visited routes, as well as backward/forward browser navigation.
+
+For the user, this results in [soft navigation](/docs/app/building-your-application/routing/linking-and-navigating#soft-navigation) (no page reloads) and faster page transitions; it also reduces unnecessary requests to the server.
+
+> **Good to know:**
+>
+> You may see the Router Cache being referred to as **Client-side Cache** or **Prefetch Cache**. While **Prefetch Cache** refers to the prefetched route segments, **Client-side Cache** refers to the whole Router cache, which includes both visited and prefetched segments.
+> This cache specifically applies to Next.js and Server Components, and is different to the browser's [bfcache](https://web.dev/bfcache/).
+
+#### Duration
+
+The cache is stored in the browser's temporary memory. Two factors determine how long the router cache lasts:
+
+- **Session**: The cache persists across [soft navigations](/docs/app/building-your-application/routing/linking-and-navigating#soft-navigation). However, it's cleared on page refresh.
+- **Automatic invalidation period**: The cache of an individual segment is automatically invalidated after a specific time. The duration depends on whether the route is [statically](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#static-rendering-default) or [dynamically](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-rendering) rendered:
+  - **Dynamically rendered**: 30 seconds
+  - **Statically Rendered**: 5 minutes
+
+While a page refresh will clear **all** cached segments, the automatic invalidation period only affects the individual segment from the time it was last accessed or created.
+
+#### Invalidation
+
+There are two ways you can invalidate the Router Cache:
+
+- Revalidating data on-demand by path with ([`revalidatePath`](/docs/app/api-reference/functions/revalidatePath)) or by cache tag with ([`revalidateTag`](/docs/app/api-reference/functions/revalidateTag)) in Server Actions, which in turn will invalidate the Router Cache.
+- Using `cookies.set` or `cookies.delete` in a Server Actions invalidates the Router Cache to prevent routes that use cookies from becoming stale (e.g. authentication).
```

Comment:
One more thing, it would be nice to mention that the Router Cache is also invalidated when navigating to/from a Pages Router's page, because it is a hard navigation (general term) [as well as `router.refresh()`](https://github.com/vercel/next.js/pull/52514/files/36c7e81d8c5cb229b4192191585ccdeff334928c#r1270623774). I was surprised when I first noticed that.

Reply:
```suggestion
- Using `cookies.set` or `cookies.delete` in a Server Actions invalidates the Router Cache to prevent routes that use cookies from becoming stale (e.g. authentication).
- Caling `router.refresh()`
```

Maybe split up the list between apis that can be called on the server and apis that can be called in the browser.

Reply:
The pages router <-> app router part I'd leave out here tbh, it's it's the same as navigating away from the page so I would document that part instead.